.github/workflows/main.yml

```diff
@@ -20,101 +20,31 @@ concurrency:
 
 jobs:
   prim-rev:
-    name: prim and rev tests
-    runs-on: windows-latest
+    name: ${{ matrix.config.label }} tests (${{ matrix.config.os }})
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - { os: windows-latest, label: misc, tests: 'test/unit/*_test.cpp test/unit/math/*_test.cpp test/unit/math/memory' }
+          - { os: ubuntu-24.04-arm, label: misc, tests: 'test/unit/*_test.cpp test/unit/math/*_test.cpp test/unit/math/memory' }
+          - { os: windows-latest, label: prim, tests: 'test/unit/math/prim' }
+          - { os: ubuntu-24.04-arm, label: prim, tests: 'test/unit/math/prim' }
+          - { os: windows-latest, label: rev, tests: 'test/unit/math/rev' }
+          - { os: ubuntu-24.04-arm, label: rev, tests: 'test/unit/math/rev' }
+          - { os: windows-latest, label: fwd, tests: 'test/unit/math/fwd' }
+          - { os: ubuntu-24.04-arm, label: fwd, tests: 'test/unit/math/fwd' }
+          - { os: windows-latest, label: 'non-fun mix', tests: 'test/unit/math/mix/core test/unit/math/mix/meta test/unit/math/mix/*_test.cpp' }
+          - { os: ubuntu-24.04-arm, label: 'non-fun mix', tests: 'test/unit/math/mix/core test/unit/math/mix/meta test/unit/math/mix/*_test.cpp' }
 
```

Comment:
I think this could be written more succintly?

```
matrix:
  os: [windows-latest, ubuntu-24.04-arm]
  tests: [{label: 'misc', paths: 'test/unit/*_test.cpp test/unit/math/*_test.cpp test/unit/math/memory'} , ... ]